### PR TITLE
Gcw 1518 dispatch from stockit

### PR DESCRIPTION
**Pull request by namrataukirde**

### CODE REVIEW

File: app/models/package.rb [hunk 1]
```diff
@@ -131,7 +135,55 @@ def is_order_id_nil?
...
+      cancel_designation(designation)
+      orders_package.dispatch
+    else
+      handle_singleton_dispatch_undispatch_with_or_without_designation(designation)
+    end
+    update_in_stock_quantity
+  end
+
+  def handle_singleton_dispatch_undispatch_with_or_without_designation(designation)
+    if is_singletone_and_has_designation?(designation) && is_stockit_sent_on_present?
```
Comment (trojanh) [inline app/models/package.rb]: typo

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 2]
```diff
@@ -75,108 +75,199 @@ def test_orders_packages(package, stockit_request, count)
...
-        test_orders_packages(package, stockit_request, 1)
-        expect(package.orders_packages.first.state).to eq('cancelled')
+        it "create new package with designation for newly created item from stockit with designation", :show_in_doc do
+          expect{
+            post :create, format: :json, package: stockit_item_params_with_designation
+          }.to change(Package, :count).by(1)
+          package = Package.where(inventory_number: stockit_item_params_with_designation[:inventory_number]).first
+          test_package_changes(package, response.status, order.code)
+          stockit_request = GoodcitySync.request_from_stockit
+          test_orders_packages(package, stockit_request, 1)
```
Comment (trojanh) [inline spec/controllers/api/v1/packages_controller_spec.rb]: above three lines can be made into a method as they are used multiple times.

File: app/models/package.rb (hunk 1 above)
Comment (namrataukirde) [inline app/models/package.rb]: Sure @trojanh , Will correct typo.

**Comment by steveyken** [comment]
Approved and merged this so we can test on staging.

**Pull request merged**